### ISSUE TITLE
Member Directory showing deleted fields to sosrt by

### ISSUE THREAD

**Issue by cerullo** [dnnsoftware/Dnn.Platform#2708]
## Description of bug
In the sort or filter by fields of the member directory it will show fields that have been deleted from the profiles/system. It is NOT filtering our fields that are marked as "deleted" 

## Steps to reproduce
List the steps to reproduce the behavior:
1. go to the settings of the member directory module
2. view the drop downs for filter/sort. 
3. fields that were deleted from the user profiles are still displaying. 

## Current result
fields that were deleted from the user profiles are still displaying. 

## Expected result
Fields that were deleted from the user profile should not be showing. 

## Affected version

* [x] 9.2.2
* [x] 9.2.1
* [x] 9.2
* [x] 9.1.1

## Affected browser
all browsers

**Comment by valadas** [comment]
I can confirm the issue on 9.4.0 RC.

Now when a profile property is deleted, it is just marked as deleted (soft delete) but stays in the database. 

Now I find it useful to still have the choice to filter by a profile property that is soft-deleted or not and I am not seing this as a bug per se and find the current behaviour more flexible than not giving that option.

Thoughs ? It would be a quick change to only show the not soft-deleted ones, I just do not right now see any good reason for such behaviour change.

**Comment by EPTamminga** [comment]
I think that soft delete = delete = gone. 
Soft delete is an option to get things back if you accidentally deleted something, or if the hard delete is not possible for data integrity reasons.
AFAIK, that is not the case for a profile property, so imo ($0,02) it should not be possible to filter on (soft) deleted properties.

**Comment by sleupold** [comment]
I agree with @EPTamminga and AFAIR the property list in the member directory module is hardcoded (not only for filtering) and it would require a major rewrite to support custom properties and suppress deleted ones.

**Comment by sleupold** [comment]
this is still a known bug
